### DISCUSSION TITLE
questions--pressure test

### DISCUSSION THREAD

**Discussion by xzyblockChain** [0xPolygon/polygon-edge#305] [Q&A] [answered]
The number of transactions on the chain per second is only more than 100. How to improve it? Did you do a pressure test?

**Comment by zivkovicmilos** [comment]
Hey @xzyblockChain,

Please open up these discussions in the discussion section.
Unless you're reporting an issue in the PSDK, there's no need for these threads to be here.

Moving to a discussion now

**Comment by zivkovicmilos** [comment, marked as answer]
What kind of setup are you running? How many nodes? What kind of transactions? What is your block gas limit?

Please provide more information

**Reply by xzyblockChain** [reply]
The pressure test is conducted for 3000 transactions per second. There are 4 nodes and the server is 4-core 8g. The test found that the number of transactions on the chain per second was only 168

**Reply by xzyblockChain** [reply]
![621f8279e0d8cc7a99c2f0b5cb4364d](https://user-images.githubusercontent.com/38971689/147069378-fc955061-b9f3-421a-bfd0-a419f21a0d8b.png)

**Reply by xzyblockChain** [reply]
Can you answer？

**Reply by mrwillis** [reply]
What's your gas limit? For a given block limit and block gas limit, there is a fundamental upper limit to the TPS
